index.html

```diff
@@ -2456,70 +2216,48 @@ <h3 id="applying">
         <p>
           The <dfn>steps for processing the <code>related_applications</code>
           member</dfn> are given by the following algorithm. The algorithm
-          takes a manifest as an argument. This algorithm returns a list of
-          application objects <var>applications</var>, which can be empty.
+          takes a <a>WebAppManifest</a> <var>manifest</var> as an argument.
+          This algorithm returns an
+          Array&lt;<a>ExternalApplicationResource</a>&gt;.
         </p>
         <ol>
-          <li>Let <var>applications</var> be an empty list.
+          <li>Let <var>A</var> be <var>manifest["related_applications"]</var>.
           </li>
-          <li>Let <var>unprocessed applications</var> be the result of calling
-          the <a>[[\GetOwnProperty]]</a> internal method of <var>manifest</var>
-          with argument "<code>related_applications</code>".
+          <li>Let <var>A'</var> be a new Array object created as if by the
+          expression [].
           </li>
-          <li>If <var>unprocessed applications</var> is an <a>array</a>, then:
+          <li>For <var>V</var> in <var>A</var>:
```

Comment:
Nit: please use for each, IDL sequences are just arrays, so it's nice to keep the language that maps to things in JS. Also, please don't use the bracket syntax, just use:
 
```HTML
For each `<var>relatedApp</var>` in `<var>manifest</var>. <a>related_applications</a>.
```

Then we get a link back to related_applications, and don't need to guess on the type.

Reply:
fixed markup above.

Reply:
That translates nicely to:

```js
for(const relatedApp of manifest.related_applications) {... do stuff ... }
```

Or, better:

```js
manifest.related_applications.forEach(relatedApp => ... do stuff... )
```

Reply:
@tobie said that we cannot use dot notation for these as it is WebIDL, have to use []

Reply:
I don’t think that’s correct. Or, if it is, I’m inclined to willfully violate that. We do the same in Payment Request. And I’ve never seen any other spec use [] notation.

Reply:
Apparently dot notation for Web IDL is not spec'ed. I went with brackets for now, but it will be easy to change (like in follow up) when dot notation gets spec'ed

Reply:
Can you open a ticket against WebIDL for that? Thanks.

Reply:
Done: https://github.com/heycam/webidl/issues/453